client/engine/chainservice/eth_chainservice.go

```diff
@@ -90,7 +90,20 @@ func newEthChainService(chain ethChain, na *NitroAdjudicator.NitroAdjudicator,
 
 	// Use a buffered channel so we don't have to worry about blocking on writing to the channel.
 	ecs := EthChainService{chain, na, naAddress, caAddress, vpaAddress, txSigner, make(chan Event, 10), logger, ctx, cancelCtx}
-	ecs.subscribeForLogs()
+	errChan, err := ecs.subscribeForLogs()
+	// TODO: Return error from chain service instead of panicking
```

Comment:
What do you see as the longer term pattern for this?

Could the chain service (and other services) forward errors to the engine (perhaps on a new `chan`), which then funnels them (again on a new `chan`) to the consuming application / RPC server?

Under that pattern, the goroutine spawned on L95 wouldn't need to exist.